### DISCUSSION TITLE
Does spin/fermyon cloud support SSE?

### DISCUSSION THREAD

**Discussion by maxwell8888** [spinframework/spin#1867] [Q&A] [answered]
I'm interested in trying spin/fermyon cloud but my usecase requires server-sent events. I can't find any reference to SSE in the spin github repo or docs. Thanks.

**Comment by itowlson** [comment, marked as answer]
It isn't possible in Spin 1.x, but it should be possible in Spin 2.x and the corresponding iteration of Fermyon Cloud.  The reason is that in Spin 1.x you have to return the entire HTTP response at once, whereas in Spin 2.x you should be able to stream it.  You'll then be able to use the server-side events protocol (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#sending_events_from_the_server) to emit events as required.

@dicej is working on streaming responses and can hopefully confirm or deny.

**Reply by dicej** [reply]
Yes, sending `text/event-stream` responses using `wasi-http` should be feasible in Spin 2.0.  We're not planning on including any SSE-specific APIs in the Spin 2.0 SDKs, but if you don't mind implementing the protocol yourself (or using a third-party library), it should work fine.
